.github/workflows/ci-lint.yml

```diff
@@ -0,0 +1,40 @@
+name: Code linting
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  lint:
```

Comment:
Should probably add a timeout to this task as well.

Reply:
It takes 60 seconds so I don't think that's necessary.

Reply:
(See here: https://github.com/fluid-project/infusion/pull/1023/checks?check_run_id=1321259265)

Reply:
The purpose would be to prevent a stalled run from hanging for a day or so. This way it forces the run to stop early. The time length can be lower than the 20 min that is set for the other tests. I believe @gtirloni picked that number as it gave time for the tests to complete successfully.

Reply:
+1 -- it's just to shield us from that because we have limited build minutes to use every month and I've seen my fair share of weird problems. It's good to have an upper limit (even if it's 20x the usual time).